crates/utils/src/utils/validation.rs

```diff
@@ -0,0 +1,84 @@
+use itertools::Itertools;
+use once_cell::sync::Lazy;
+use regex::Regex;
+use url::Url;
+
+static VALID_ACTOR_NAME_REGEX: Lazy<Regex> =
+  Lazy::new(|| Regex::new(r"^[a-zA-Z0-9_]{3,}$").expect("compile regex"));
+static VALID_POST_TITLE_REGEX: Lazy<Regex> =
+  Lazy::new(|| Regex::new(r".*\S{3,}.*").expect("compile regex"));
+static VALID_MATRIX_ID_REGEX: Lazy<Regex> = Lazy::new(|| {
+  Regex::new(r"^@[A-Za-z0-9._=-]+:[A-Za-z0-9.-]+\.[A-Za-z]{2,}$").expect("compile regex")
+});
+// taken from https://en.wikipedia.org/wiki/UTM_parameters
+static CLEAN_URL_PARAMS_REGEX: Lazy<Regex> = Lazy::new(|| {
+  Regex::new(r"^utm_source|utm_medium|utm_campaign|utm_term|utm_content|gclid|gclsrc|dclid|fbclid$")
+    .expect("compile regex")
+});
+
+fn has_newline(name: &str) -> bool {
+  name.contains('\n')
+}
```

Comment:
Why is this necessary? Should be caught by other regexes already.

Reply:
I tested out the post_title regex, and it will pass with a `\n`, which is probably why this was added.

Reply:
Weird, shouldnt be necessary based on my understanding of regex docs but okay.